include/libcyphal/transport/can/transport.hpp

```diff
@@ -0,0 +1,219 @@
+/// Copyright Amazon.com Inc. or its affiliates.
+/// SPDX-License-Identifier: MIT
+
+#ifndef LIBCYPHAL_TRANSPORT_CAN_TRANSPORT_HPP_INCLUDED
+#define LIBCYPHAL_TRANSPORT_CAN_TRANSPORT_HPP_INCLUDED
+
+#include "media.hpp"
+#include "libcyphal/transport/transport.hpp"
+#include "libcyphal/transport/multiplexer.hpp"
+
+#include <canard.h>
+
+#include <algorithm>
+
+namespace libcyphal
+{
+namespace transport
+{
+namespace can
+{
+
+class ICanTransport : public ITransport
+{};
+
+namespace detail
+{
+class TransportImpl final : public ICanTransport
+{
+public:
+    TransportImpl(cetl::pmr::memory_resource&            memory,
+                  libcyphal::detail::VarArray<IMedia*>&& media_array,
+                  const CanardNodeID                     canard_node_id)
+        : memory_{memory}
+        , media_array_{std::move(media_array)}
+        , canard_instance_{canardInit(canardMemoryAllocate, canardMemoryFree)}
+    {
+        canard_instance_.user_reference = this;
+        canard_instance_.node_id        = canard_node_id;
+    }
+
+    // MARK: ICanTransport
+
+    // MARK: ITransport
+
+    CETL_NODISCARD cetl::optional<NodeId> getLocalNodeId() const noexcept override
+    {
+        return canard_instance_.node_id > CANARD_NODE_ID_MAX ? cetl::nullopt
+                                                             : cetl::make_optional(canard_instance_.node_id);
+    }
+    CETL_NODISCARD ProtocolParams getProtocolParams() const noexcept override
+    {
+        const auto min_mtu = reduceMediaInto(std::numeric_limits<std::size_t>::max(),
+                                             [](auto&& mtu, IMedia& media) { mtu = std::min(mtu, media.getMtu()); });
+        return ProtocolParams{1 << CANARD_TRANSFER_ID_BIT_LENGTH, min_mtu, CANARD_NODE_ID_MAX + 1};
+    }
+
+    CETL_NODISCARD Expected<UniquePtr<IMessageRxSession>, AnyError> makeMessageRxSession(
+        const MessageRxParams&) override
+    {
+        return NotImplementedError{};
+    }
+    CETL_NODISCARD Expected<UniquePtr<IMessageTxSession>, AnyError> makeMessageTxSession(
+        const MessageTxParams&) override
+    {
+        return NotImplementedError{};
+    }
+    CETL_NODISCARD Expected<UniquePtr<IRequestRxSession>, AnyError> makeRequestRxSession(
+        const RequestRxParams&) override
+    {
+        return NotImplementedError{};
+    }
+    CETL_NODISCARD Expected<UniquePtr<IRequestTxSession>, AnyError> makeRequestTxSession(
+        const RequestTxParams&) override
+    {
+        return NotImplementedError{};
+    }
+    CETL_NODISCARD Expected<UniquePtr<IResponseRxSession>, AnyError> makeResponseRxSession(
+        const ResponseRxParams&) override
+    {
+        return NotImplementedError{};
+    }
+    CETL_NODISCARD Expected<UniquePtr<IResponseTxSession>, AnyError> makeResponseTxSession(
+        const ResponseTxParams&) override
+    {
+        return NotImplementedError{};
+    }
+
+    // MARK: IRunnable
```

Comment:
Who's Mark? Do I know him?

Reply:
![image](https://github.com/OpenCyphal-Garage/libcyphal/assets/3298404/e86944a7-5af0-47bf-87b0-c52362fb0314)

Reply:
It looks like this in CLion:
<img width="1132" alt="Screenshot 2024-04-23 at 12 46 01" src="https://github.com/OpenCyphal-Garage/libcyphal/assets/2915466/75d4eee2-293a-46be-9f26-e5109dfc6b6f">